sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/state/StreamStreamJoinStateHelper.scala

```diff
@@ -68,11 +68,11 @@ object StreamStreamJoinStateHelper {
     val newHadoopConf = session.sessionState.newHadoopConf()
 
     val manager = new StateSchemaCompatibilityChecker(providerIdForKeyToNumValues, newHadoopConf)
-    val (keySchema, _) = manager.readSchemaFile()
+    val keySchema = manager.readSchemaFile().head.keySchema
 
     val manager2 = new StateSchemaCompatibilityChecker(providerIdForKeyWithIndexToValue,
```

Comment:
can you add a comment on what this is doing? Why are we reading the keySchema from one manager and valueSchema from another?

Reply:
Done